frame/staking/src/pallet/mod.rs

```diff
@@ -816,10 +824,15 @@ pub mod pallet {
 
 				// NOTE: ledger must be updated prior to calling `Self::weight_of`.
 				Self::update_ledger(&controller, &ledger);
+
 				// update this staker in the sorted list, if they exist in it.
-				if T::SortedListProvider::contains(&stash) {
-					T::SortedListProvider::on_update(&stash, Self::weight_of(&ledger.stash));
-					debug_assert_eq!(T::SortedListProvider::sanity_check(), Ok(()));
+				if T::VoterList::contains(&stash) {
+					T::VoterList::on_update(&stash, Self::weight_of(&ledger.stash));
+					debug_assert_eq!(T::VoterList::sanity_check(), Ok(()));
+				}
+				if T::TargetList::contains(&stash) {
+					T::TargetList::on_update(&stash, Self::weight_of(&ledger.stash));
+					debug_assert_eq!(T::VoterList::sanity_check(), Ok(()));
```

Comment:
```suggestion
					debug_assert_eq!(T::TargetList::sanity_check(), Ok(()));
```
We also can remove the debug assert here and above and instead put `Self::sanity_check_list_providers();` after these two if statements